### PR TITLE
Add unit tests for common.go in pkg/util

### PR DESCRIPTION
**Pull request by NishantBansal2003**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
The test case coverage for common.go has been increased to 100%.
**Which issue(s) this PR fixes**:
Ref #5235 

**Special notes for your reviewer**:
To verify the changes in the pkg/util directory run the following commands:
```
go test ./... -coverprofile=coverage.out
go tool cover -html=coverage.out -o coverage.html
open coverage.html
```
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by NishantBansal2003** [comment]
Hey @XiShanYongYe-Chang added 100% test coverage in `pkg/util/common.go`. PTAL...

**Comment by XiShanYongYe-Chang** [comment]
Thanks @NishantBansal2003 
/assign

**Comment by NishantBansal2003** [comment]
Hey @XiShanYongYe-Chang, I have some doubts I hope you can resolve:
1. Suppose there is a helper function used to test the functionality of another function, but not all aspects of the helper function are utilized (and therefore remain untested). Should we create separate tests specifically for this helper function, or is it unnecessary to test it?
2. I've read some of your reviews on PRs where you mentioned, `Although the current test content improves the file coverage, the test cases have no real meaning. Additionally, the maintenance cost is increased.` Could you explain this further so I can avoid making similar mistakes?

**Comment by XiShanYongYe-Chang** [comment]
> Suppose there is a helper function used to test the functionality of another function, but not all aspects of the helper function are utilized (and therefore remain untested). Should we create separate tests specifically for this helper function, or is it unnecessary to test it?

I'm afraid I answered wrong. do you have any specific examples?

> I've read some of your reviews on PRs where you mentioned, Although the current test content improves the file coverage, the test cases have no real meaning. Additionally, the maintenance cost is increased. Could you explain this further so I can avoid making similar mistakes?

The functions to be tested should have clear purposes and the logic can be tested. For example, different test paths can be used to form different test cases. If the test case requires the same maintenance cost as the function under test, then I don't think the test is needed.

File: pkg/util/common_test.go [hunk 1]
```diff
@@ -54,51 +54,79 @@ var (
...
+			isRemovedExpected := reflect.DeepEqual(removed, tt.expectedRemoved)
+			if !isAddedExpected && !isRemovedExpected {
+				t.Errorf("added = %v want %v, removed = %v want %v", added, tt.expectedAdded, removed, tt.expectedRemoved)
+			}
+			if !isAddedExpected {
+				t.Errorf("added = %v, want %v", added, tt.expectedAdded)
+			}
+			if !isRemovedExpected {
+				t.Errorf("removed = %v want %v", removed, tt.expectedRemoved)
+			}
```
Comment (XiShanYongYe-Chang) [inline pkg/util/common_test.go:127]: How about update like this:
```diff
                        added, removed := DiffKey(tt.previous, tt.current)
-                       isAddedExpected := reflect.DeepEqual(added, tt.expectedAdded)
-                       isRemovedExpected := reflect.DeepEqual(removed, tt.expectedRemoved)
-                       if !isAddedExpected && !isRemovedExpected {
-                               t.Errorf("added = %v want %v, removed = %v want %v", added, tt.expectedAdded, removed, tt.expectedRemoved)
-                       }
-                       if !isAddedExpected {
+                       if !reflect.DeepEqual(added, tt.expectedAdded) {
                                t.Errorf("added = %v, want %v", added, tt.expectedAdded)
                        }
-                       if !isRemovedExpected {
+                       if !reflect.DeepEqual(removed, tt.expectedRemoved) {
                                t.Errorf("removed = %v want %v", removed, tt.expectedRemoved)
                        }
```

File: pkg/util/common_test.go [hunk 2]
```diff
@@ -111,15 +139,26 @@ func TestStringerJoin(t *testing.T) {
...
+	})
+	t.Run("empty keys", func(t *testing.T) {
+		var mcs map[string]sigMultiCluster
+		got := Keys(mcs)
+		var expect []string
+		sort.Strings(got)
+		if !reflect.DeepEqual(got, expect) {
+			t.Errorf("got = %v, want %v", got, expect)
+		}
+	})
```
Comment (XiShanYongYe-Chang) [inline pkg/util/common_test.go:163]: How about rewriting this test as tabel drive as well? Just like your rewrite above.

Comment (NishantBansal2003) [reply pkg/util/common_test.go]: I thought the same. Now I'm doing it...

**Comment by NishantBansal2003** [comment]
> I'm afraid I answered wrong. do you have any specific examples?

Yeah like this -
https://github.com/karmada-io/karmada/blob/6e9136d5e99dcd7bcd7ae620bfb51ab668a12210/pkg/util/testing/mock_responder.go#L32-L36
Is used to test `TestConnectCluster` in https://github.com/karmada-io/karmada/blob/master/pkg/util/proxy/proxy_test.go
But other functions like -
https://github.com/karmada-io/karmada/blob/6e9136d5e99dcd7bcd7ae620bfb51ab668a12210/pkg/util/testing/mock_responder.go#L39-L48
Has not been used, Hence remain untested. So should I create separate tests specifically for this(untested) function, or is it unnecessary to test it?

**Comment by NishantBansal2003** [comment]
PTAL...

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm
/approve

**Comment by XiShanYongYe-Chang** [comment]
> Has not been used, Hence remain untested. So should I create separate tests specifically for this(untested) function, or is it unnecessary to test it?

I understand that this is part of the test framework, no need to test them.

**Pull request merged**